quesma/quesma/schema_transformer.go

```diff
@@ -789,3 +790,42 @@ func (s *SchemaCheckPass) Transform(queries []*model.Query) ([]*model.Query, err
 	}
 	return queries, nil
 }
+
+func (s *SchemaCheckPass) applyMatchOperator(indexSchema schema.Schema, query *model.Query) (*model.Query, error) {
+
+	visitor := model.NewBaseVisitor()
+
+	var err error
+
+	visitor.OverrideVisitInfix = func(b *model.BaseExprVisitor, e model.InfixExpr) interface{} {
+		lhs, ok := e.Left.(model.ColumnRef)
+		rhs, ok2 := e.Right.(model.LiteralExpr)
+
+		if ok && ok2 && e.Op == model.MatchOperator {
+			field, found := indexSchema.ResolveFieldByInternalName(lhs.ColumnName)
+			if !found {
+				logger.Error().Msgf("Field %s not found in schema for table %s, should never happen here", lhs.ColumnName, query.TableName)
+			}
+			switch field.Type.String() {
+			case schema.QuesmaTypeInteger.Name, schema.QuesmaTypeLong.Name, schema.QuesmaTypeUnsignedLong.Name:
+				return model.NewInfixExpr(lhs, "=", model.NewLiteral(rhs.Value.(string)))
```

Comment:
What about other data types? data, float?

Reply:
The Elastic's documentation doesn't specify this behavior (and the code itself is convoluted), but empirically testing:
- float: doesn't work (tried `AvgTicketPrice:123.12` filter)
- boolean: works, therefore added it to switch
- date: `date_field:[DATE_VALUE]` doesn't work; you can click to filter by it in Kibana, but then it emits different filter than `match`
- keyword: `keyword_field:[KEYWORD_VALUE]` emits a different filter than `match`

I did not test other types. Therefore, I think it's OK for us to support only those types right now.